File: backend/app/schemas/ticket.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel

class TicketStatus(str, Enum):
    open = "open"
    in_progress = "in_progress"
    closed = "closed"

class TicketPriority(str, Enum):
    low = "low"
    medium = "medium"
    high = "high"

class TicketCreate(BaseModel):
    title: str
    description: str
    priority: TicketPriority = TicketPriority.medium


class TicketUpdate(BaseModel):
    status: TicketStatus


class TicketResponse(BaseModel):
    id: int
    title: str
    description: str
    status: TicketStatus
    priority: TicketPriority
    user_id: int
    created_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend/src/types/ticket.ts
export type TicketStatus =
  | "open"
  | "in_progress"
  | "closed";

export type TicketPriority =
  | "low"
  | "medium"
  | "high";


export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: string;
  priority: string;
  created_at: string;
}